lib/backpex/field.ex

```diff
@@ -1,9 +1,109 @@
 # credo:disable-for-this-file Credo.Check.Refactor.CyclomaticComplexity
 defmodule Backpex.Field do
-  @moduledoc ~S'''
+  @config_schema [
+    module: [
+      doc: "The field module.",
+      type: :atom,
+      required: true
+    ],
+    label: [
+      doc: "The field label.",
+      type: :string,
+      required: true
+    ],
+    render: [
+      doc: "A function to overwrite the template used . It should take `assigns` and return a HEEX template.",
+      type: {:fun, 1}
+    ],
+    render_form: [
+      doc: "A function to overwrite the template used in forms. It should take `assigns` and return a HEEX template.",
+      type: {:fun, 1}
+    ],
+    custom_alias: [
+      doc: "A custom alias for the field.",
+      type: :atom
+    ],
+    align: [
+      doc: "Align the fields of a resource in the index view.",
+      type: {:in, [:left, :center, :right]}
+    ],
+    align_label: [
+      doc: "Align the labels of the fields in the edit view.",
+      type: {:in, [:top, :center, :bottom]}
+    ],
+    searchable: [
+      doc: "Define wether this field should be searchable on the index view.",
+      type: :boolean
+    ],
+    orderable: [
+      doc: "Define wether this field should be orderable on the index view.",
+      type: :boolean
+    ],
+    visible: [
+      doc: "Function to change the visibility of a field. Receives the assigns and has to return a boolean.",
+      type: {:fun, 1}
+    ],
+    panel: [
+      doc: "Group field into panel. Also see the [panels](/guides/authorization/panels.md) guide.",
+      type: :atom
+    ],
+    index_editable: [
+      doc: """
+      Define wether this field should be editable on the index view. Also see the
+      [index edit](/guides/authorization/index-edit.md) guide.
+      """,
+      type: :boolean
+    ],
+    index_column_class: [
+      doc: """
+      Add additional class(es) to the index column.
+      In case of a function it takes the `assigns` and should return a string.
+      """,
+      type: {:or, [:string, {:fun, 1}]}
+    ],
+    select: [
+      doc: """
+      Define a dynamic select query expression for this field.
+
+      ### Example
+
+          full_name: %{
+            module: Backpex.Fields.Text,
+            label: "Full Name",
+            select: dynamic([user: u], fragment("concat(?, ' ', ?)", u.first_name, u.last_name)),
+          }
+      """,
+      type: {:struct, Ecto.Query.DynamicExpr}
```

Comment:
I get the following warning when generating the docs:

```
Generating docs...
     warning: documentation references module "Ecto.Query.DynamicExpr" but it is hidden
     │
 134 │   @callback render_index_form(assigns :: map()) :: %Phoenix.LiveView.Rendered{}
     │   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     │
     └─ lib/backpex/field.ex:134: Backpex.Field (module)
```

But maybe thats not a problem?

Reply:
We may use `skip_undefined_reference_warnings_on` to suppress the warning (?)

Reply:
👍 But had to use `skip_code_autolink_to` to suppress the warning.